code/datums/weather/weather.dm

```diff
@@ -128,15 +131,18 @@
 /datum/weather/proc/start()
 	if(stage >= MAIN_STAGE)
 		return
+	SEND_GLOBAL_SIGNAL(COMSIG_WEATHER_START(type))
 	stage = MAIN_STAGE
 	update_areas()
-	for(var/M in GLOB.player_list)
-		var/turf/mob_turf = get_turf(M)
-		if(mob_turf && (mob_turf.z in impacted_z_levels))
+	for(var/z_level in impacted_z_levels)
+		for(var/mob/player as anything in SSmobs.clients_by_z_level[z_level])
```

Comment:
```suggestion
		for(var/mob/player as anything in SSmobs.clients_by_zlevel[z_level])
```

Reply:
my bad i thought there was another underscore there

Reply:
it's all good